frontend/talentsearch/src/js/components/applicantProfile/ExperienceAndSkills.tsx

```diff
@@ -0,0 +1,365 @@
+        data-h2-padding="b(top-bottom, xxs) b(right-left, xs)"
+      >
+        {experiences.map((experience) => (
+          <ExperienceAccordion key={experience.id} experience={experience} />
+        ))}
+      </div>
+    </div>
+  );
+};
+export interface ExperienceAndSkillsProps {
+  experiences?: Experience[];
+}
+
+const ExperienceAndSkills: React.FunctionComponent<
+  ExperienceAndSkillsProps
+> = ({ experiences }) => {
+  const intl = useIntl();
+  const compareByDate = (e1: any, e2: any) =>
+    e1.endDate && e2.endDate
+      ? new Date(e2.endDate).getTime() - new Date(e1.endDate).getTime()
+      : 0;
+  const awardExperiences =
+    experiences
+      ?.filter((experience) => isAwardExperience(experience))
+      .map(
+        (award: AwardExperience) =>
+          ({
+            ...award,
+            endDate: award.awardedDate,
+          } as AwardExperience),
+      )
+      .sort(compareByDate) || [];
+  const communityExperiences =
+    experiences
+      ?.filter((experience) => isCommunityExperience(experience))
+      .sort(compareByDate) || [];
+  const educationExperiences =
+    experiences
+      ?.filter((experience) => isEducationExperience(experience))
+      .sort(compareByDate) || [];
+  const personalExperiences =
+    experiences
+      ?.filter((experience) => isPersonalExperience(experience))
+      .sort(compareByDate) || [];
+  const workExperiences =
+    experiences
+      ?.filter((experience) => isWorkExperience(experience))
+      .sort(compareByDate) || [];
+
+  const links = [
+    {
+      href: "/",
+      title: intl.formatMessage({
+        defaultMessage: "Personal",
+        description: "Title for personal experience form button.",
+      }),
+      icon: <LightBulbIcon style={{ width: "1.5rem" }} />,
+    },
+    {
+      href: "/",
+      title: intl.formatMessage({
+        defaultMessage: "Community",
+        description: "Title for community experience form button.",
+      }),
+      icon: <UserGroupIcon style={{ width: "1.5rem" }} />,
+    },
+    {
+      href: "/",
+      title: intl.formatMessage({
+        defaultMessage: "Work",
+        description: "Title for work experience form button.",
+      }),
+      icon: <BriefcaseIcon style={{ width: "1.5rem" }} />,
+    },
+    {
+      href: "/",
+      title: intl.formatMessage({
+        defaultMessage: "Education",
+        description: "Title for education experience form button.",
+      }),
+      icon: <BookOpenIcon style={{ width: "1.5rem" }} />,
+    },
+    {
+      href: "/",
+      title: intl.formatMessage({
+        defaultMessage: "Award",
+        description: "Title for award experience form button.",
+      }),
+      icon: <StarIcon style={{ width: "1.5rem" }} />,
+    },
+  ];
+
+  const allExperiences = [
+    ...awardExperiences,
+    ...communityExperiences,
+    ...educationExperiences,
+    ...personalExperiences,
+    ...workExperiences,
+  ];
+
+  const sortedByDate = allExperiences.sort(compareByDate);
```

Comment:
I think the logic is supposed to be a little more complex.  Was it enddate and then startdate in case two have the same end date?

Reply:
Ya you're right I made a fix according to what [Gray stated](https://github.com/GCTC-NTGC/gc-digital-talent/issues/1926#issuecomment-1032802495).